butane_core/src/migrations/fsmigrations.rs

```diff
@@ -95,7 +98,14 @@ impl FsMigration {
         self.fs
             .write(&path)?
             .write_all(contents)
-            .map_err(|e| e.into())
+            .map_err(<std::io::Error as Into<Error>>::into)?;
+        if contents[contents.len() - 1] != b'\n' {
+            self.fs
```

Comment:
This additional open is inefficient, but allows the second commit to be optional.
Still a good idea to have this to be sure the file has EOL at EOF.  Or could become an error to ensure we never forget to append the EOL before calling `write_contents`.

Reply:
Two questions here:
1. Rather than re-opening the file, why not just assign the first writer to a local variable and do the check and second write before letting it go?
2. If you're auto-appending a newline in this function, why bother with all the concacts at the call sites?

Reply:
As these writes are not IMO a performance-sensitive path, my intention was to only use the first commit, and discard the second.  That involved two opens, which isnt performant, but it is clear and simple.

There is no need for the second commit which uses concats, which I think add unnecessary complication.  Maybe there is a way to do the append neater in the callers, but I couldnt see an easy one.  https://github.com/serde-rs/json/issues/315 shows how it can be done.  https://github.com/serde-rs/json/issues/746 would require a lot more restructuring.

Using a variable in `write_contents()` avoids the second file open, so I am happy to go with that, and drop the second commit.